lib/src/importer/node_package.dart

```diff
@@ -0,0 +1,389 @@
+// Copyright 2024 Google Inc. Use of this source code is governed by an
+// MIT-style license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import 'package:cli_pkg/js.dart';
+import 'package:collection/collection.dart';
+import 'package:sass/src/util/map.dart';
+import 'package:sass/src/util/nullable.dart';
+
+import '../importer.dart';
+import './utils.dart';
+import 'dart:convert';
+import '../io.dart';
+import 'package:path/path.dart' as p;
+
+/// An [Importer] that resolves `pkg:` URLs using the Node resolution algorithm.
+class NodePackageImporter extends Importer {
+  /// The starting path for canonicalizations without a containing URL.
+  late final String _entryPointPath;
+
+  /// Creates a Node package importer with the associated entry point.
+  NodePackageImporter(String? entryPointPath) {
+    if (entryPointPath == null) {
+      throw "The Node package importer cannot determine an entry point "
+          "because `require.main.filename` is not defined. "
+          "Please provide an `entryPointPath` to the `NodePackageImporter`.";
+    }
+    if (isBrowser) {
+      throw "The Node package importer cannot be used without a filesystem.";
+    }
+    _entryPointPath = p.absolute(entryPointPath);
+  }
+
+  /// The set of file extensions that Sass can parse. NodePackageImporter will
```

Comment:
[Separate the first sentence of a doc comment into its own paragraph](https://dart.dev/effective-dart/documentation#do-separate-the-first-sentence-of-a-doc-comment-into-its-own-paragraph)